spec/jobs/inherit_permissions_job_spec.rb

```diff
@@ -2,92 +2,199 @@
+      it 'copies permissions to its contained files' do
+        resource.permission_manager.acl.grant(:edit).to(user2).save
+        expect(resource.edit_users).to match_array [user.to_s, user2.to_s]
+
+        # files have the depositor as the edit user to begin with
+        Hyrax.query_service.custom_queries.find_child_filesets(resource: resource).each do |file|
+          expect(file.edit_users).to match_array [user.to_s]
+        end
 
-    it 'copies permissions to its contained files' do
-      # files have the depositor as the edit user to begin with
-      expect(work.file_sets.first.read_groups).to eq []
+        described_class.perform_now(resource, use_valkyrie: true)
 
-      described_class.perform_now(work)
-      work.reload.file_sets.each do |file|
-        expect(file.read_groups).to match_array ["my_read_group"]
-        expect(file.edit_users).to match_array [user.to_s]
+        # files have both edit users from parent resource
+        Hyrax.query_service.custom_queries.find_child_filesets(resource: resource).each do |file|
+          expect(file.edit_users).to match_array [user.to_s, user2.to_s]
+        end
       end
     end
-  end
 
-  context "when edit groups change" do
-    let(:name) { 'my_edit_group' }
-    let(:type) { 'group' }
-    let(:access) { 'edit' }
+    context "when people should be removed" do
+      context "when use_valkyrie is true" do
+        it 'copies permissions to its contained files' do
+          file_set.permission_manager.acl.grant(:edit).to(user2).save
+          # files have the depositor as the edit user to begin with
+          expect(resource.edit_users).to match_array [user.to_s]
+
+          # files have the depositor and extra user as the edit users to begin with
+          Hyrax.query_service.custom_queries.find_child_filesets(resource: resource).each do |file|
+            expect(file.edit_users).to match_array [user.to_s, user2.to_s]
+          end
+
+          described_class.perform_now(resource, use_valkyrie: true)
+
+          # files have single edit user from parent resource
+          Hyrax.query_service.custom_queries.find_child_filesets(resource: resource).each do |file|
+            expect(file.edit_users).to match_array [user.to_s]
+          end
+        end
+      end
+    end
+
+    context "when read people change" do
+      it 'copies permissions to its contained files' do
+        resource.permission_manager.acl.grant(:read).to(user2).save
+        expect(resource.edit_users).to match_array [user.to_s]
+
+        # files have no read users to begin with
+        Hyrax.query_service.custom_queries.find_child_filesets(resource: resource).each do |file|
+          expect(file.read_users.to_a).to be_empty
+        end
+
+        described_class.perform_now(resource, use_valkyrie: true)
+
+        # files have the extra user as the read user
+        Hyrax.query_service.custom_queries.find_child_filesets(resource: resource).each do |file|
```

Comment:
In reading this spec, it would be possible that we have an empty array, which would mean the spec is not testing anything.

It may be helpful to capture the `Hyrax.query_service.custom_queries.find_child_filesets(resource: resource)` as a variable and say that you expect it to have a specific count/size.